src/Entity/Question.php

```diff
@@ -381,6 +390,17 @@ public function preSave(EntityStorageInterface $storage) {
     }
   }
 
+  /**
+   * {@inheritdoc}
+   */
+  public static function preCreate(EntityStorageInterface $storage, array &$values) {
+    $asker_ip_address = \Drupal::request()->getClientIP();
+    if(!in_array($asker_ip_address, ['::1', '127.0.0.1', '0.0.0.0'])) {
```

Comment:
This seems like a good idea but filter_var() with FILTER_VALIDATE_IP would probably be a better way to do this and it is already possibly done for the value getClientIP() returns. With a quick look I'm not sure what kind of flags are used that check, however.

Reply:
Added the filter_var(), there was two good rules for filtering out the reserved and private ip addresses. :+1: